yt_dlp/extractor/iprima.py

```diff
@@ -115,14 +133,18 @@ def _real_extract(self, url):
 
         webpage = self._download_webpage(url, video_id)
 
-        title = self._html_search_meta(
+        title = self._html_extract_title(webpage)
+        title = title or self._html_search_meta(
             ['og:title', 'twitter:title'],
             webpage, 'title', default=None)
 
-        video_id = self._search_regex((
-            r'productId\s*=\s*([\'"])(?P<id>p\d+)\1',
-            r'pproduct_id\s*=\s*([\'"])(?P<id>p\d+)\1'),
-            webpage, 'real id', group='id')
+        video_id = self._search_regex(
+            (
+                r'productId\s*=\s*([\'"])(?P<id>p\d+)\1',
+                r'pproduct_id\s*=\s*([\'"])(?P<id>p\d+)\1',
+                r'([\'"]?)videoLength\1\s*:.*?([\'"]?)videoPlayId\2\s*:\s*(["\'])(?P<id>p\d+)\3',
```

Comment:
avoid .*?

Reply:
It has been observed that the relevant video id is available after the first presence of attribute videoLength in attribute videoPlayId.

Currently, there is an attribute in between those 2 ones. I wanted the regex to be at least a bit flexible to avoid relying too much on the page's current structure.

I don't really like the way the id is extracted, however without resorting to interpreting Javascript, I don't see a better way.

Can you please suggest a suitable alternative here?

Reply:
The third looks a part of JSON; correct?
If so, extract the whole JSON and traverse it, as a fallback

Reply:
Unfortunately, it's not a JSON, rather it's Javascript code also with variables

`(function(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,A,B,C,D,E,F,G,H,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y,Z,_,$,aa,ab,ac,ad,ae,af,ag,ah,ai,aj,ak,al,am,an,ao){return {data:{$lXB3bqXEe3:{content:{id:"ef6cf9de-c980-4443-92e4-17fe8bccd45c",drupalId:"2134783",type:"episode",images:{"1x1":a,"3x5":a,"16x9":"https:\u002F\u002Fd31b9s05ygj54s.cloudfront.net\u002Fprima-plus\u002Fimage\u002Fvideo-ef6cf9de-c980-4443-92e4-17fe8bccd45c-16x9.jpeg"},title:"Partička (92)",webUrl:Y`[...]

Reply:
Oh sorry, but then looks like it's eligible for `_search_nuxt_data`
https://github.com/yt-dlp/yt-dlp/blob/8e9fe43cd393e69fa49b3d842aa3180c1d105b8f/yt_dlp/extractor/common.py#L1662